packages/grafana-data/src/events/types.ts

```diff
@@ -88,13 +93,11 @@ export interface EventBus {
    * Remove all event subscriptions
    */
   removeAllListeners(): void;
+}
 
-  /**
-   * Returns a new bus scoped that knows where it exists in a heiarchy
-   *
-   * @internal -- This is included for internal use only should not be used directly
-   */
-  newScopedBus(key: string, filter: EventFilterOptions): EventBus;
+export interface EventBusWithFiltering<T extends EventFilterOptions = EventFilterOptions> extends EventBus {
+  readonly filterConfig: T;
+  setFilterConfig(config: Partial<T>): void;
```

Comment:
Could this be solved by recreating the scopedEvent bus instead of offering the possibility to change filter on the fly?